robot/packages/cloud_package_manager.go

```diff
@@ -478,15 +479,31 @@ func (m *cloudManager) downloadFileFromGCSURL(
 		return checksum, contentType, err
 	}
 
-	outHash := base64.StdEncoding.EncodeToString(hash.Sum(nil))
-	if outHash != checksum {
+	checksumBytes, err := base64.StdEncoding.DecodeString(checksum)
+	if err != nil {
+		return "", "", errors.Wrap(err, "failed to decode expected checksum")
+	}
+
+	trimmedChecksumBytes := trimLeadingZeroes(checksumBytes)
+	trimmedOutHashBytes := trimLeadingZeroes(hash.Sum(nil))
+
+	if !bytes.Equal(trimmedOutHashBytes, trimmedChecksumBytes) {
 		utils.UncheckedError(os.Remove(downloadPath))
-		return checksum, contentType, errors.Errorf("download did not match expected hash %s != %s", checksum, outHash)
+		return checksum, contentType, errors.Errorf("download did not match expected hash %x != %x", trimmedChecksumBytes, trimmedOutHashBytes)
 	}
 
 	return checksum, contentType, nil
 }
 
+func trimLeadingZeroes(data []byte) []byte {
+	for i, b := range data {
+		if b != 0x00 {
+			return data[i:]
+		}
+	}
+	return data // return the original slice if it's all zeroes or empty
```

Comment:
I'm guessing that the all zeroes case will be a single 0x0, but it's not specified in their docs

GCS is sending down base64 of a crc32c value (per docs [here](https://cloud.google.com/storage/docs/json_api/v1/objects) and [here](https://cloud.google.com/storage/docs/hashes-etags#json-api))

you could compare the uint32 hash value directly + not have to worry about leading zeroes

otherwise, you should turn all zeroes to 0x0 (unless you think leading zeroes are included in the all-zeroes case)

Reply:
The reason I added this trim func along with comparing []byte values is because when I compared with bytes.Equal it still had the leading zeroes

Reply:
OH wait I think I understand your comment. I think I realized the same thing reading it over again today. I changed the logic